src/server/components/dashboard/controllers.ts

```diff
@@ -142,8 +142,10 @@ export async function listsController(req: Request, res: Response, next: NextFun
     if (req.isUnauthenticated()) {
       return res.redirect(authRoutes.logout);
     }
-
-    const lists = await req.user?.getLists();
+    // TODO: Object.hasOwn is recommended but is not currently supported by tsc.
+    // eslint-disable-next-line no-prototype-builtins
+    const orderBy = req.query?.hasOwnProperty("administrators") ? req.query : undefined;
```

Comment:
`calculateSortOrder` should be done in this controller, closest to the request and the query. 

```
const { query } = req;
const orderBy = calculateSortOrder(query);
// alternatively, you can use middleware to calculate sort order and put it in res.locals. 

const lists = req.user?.getLists(orderBy);

```